pkg/ingester/instance.go

```diff
@@ -294,8 +294,10 @@ func (i *instance) createStream(ctx context.Context, pushReqStream logproto.Stre
 		err = i.streamCountLimiter.AssertNewStreamAllowed(i.instanceID)
 	}
 
+	retentionHours := i.limiter.limits.RetentionHours(i.instanceID, labels)
+
 	if err != nil {
-		return i.onStreamCreationError(ctx, pushReqStream, err, labels)
+		return i.onStreamCreationError(ctx, pushReqStream, err, labels, retentionHours)
```

Comment:
nit: This err handler looks a bit off-place. What about getting the retentionHours after getting the labels?
```go
        labels, err := syntax.ParseLabels(pushReqStream.Labels)
	if err != nil {
            ...
 	}

	retentionHours := i.limiter.limits.RetentionHours(i.instanceID, labels)

	if record != nil {
		err = i.streamCountLimiter.AssertNewStreamAllowed(i.instanceID)
                return i.onStreamCreationError(ctx, pushReqStream, err, labels, retentionHours)
	}
```

Reply:
are you sure? I'm doing it immediately after parsing the labels.